app/services/newsletter-service.js

```diff
@@ -1,94 +1,120 @@
 export default class NewsletterService extends Service {
   @service store;
   @service toaster;
   @service intl;
   @service formatter;
   @service currentSession;
 
-  async createCampaign(agenda, meeting) {
-    try {
-      const result = await ajax({
-        method: 'POST',
-        url: `/newsletter/createCampaign?agendaId=${agenda.get('id')}`,
-      });
-
-      const mailCampaign = this.store.createRecord('mail-campaign', {
-        campaignId: result.body.campaign_id,
-        campaignWebId: result.body.campaign_web_id,
-        archiveUrl: result.body.archive_url,
-      });
-
-      await mailCampaign.save();
-      const reloadedMeeting = await this.store.findRecord(
-        'meeting',
-        meeting.id,
-        {
-          reload: true,
-        }
-      );
-      reloadedMeeting.set('mailCampaign', mailCampaign);
-      await reloadedMeeting.save();
-      return mailCampaign;
-    } catch (error) {
-      console.warn('An exception ocurred: ', error);
+  async createCampaign( meeting) {
+    const endpoint = `/newsletter/mail-campaign`;
+    const body = {
+      data: {
+        meetingId: meeting.id,
+      },
+    };
+    const response = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/vnd.api+json',
+      },
+      body: JSON.stringify(body),
+    });
+    if (!response.ok) {
       this.toaster.error(
         this.intl.t('error-create-newsletter'),
         this.intl.t('warning-title')
       );
-      return null;
+      throw new Error('An exception ocurred: ' + response.error);
     }
+    const result = await response.json();
+    const mailCampaign = this.store.createRecord('mail-campaign', {
+      campaignId: result.data.id,
+      campaignWebId: result.data.attributes.webId,
+      archiveUrl: result.data.attributes.archiveUrl,
+    });
+
+    await mailCampaign.save();
+    const reloadedMeeting = await this.store.findRecord(
+      'meeting',
+      meeting.id,
+      {
+        reload: true,
+      }
+    );
+    reloadedMeeting.mailCampaign = mailCampaign;
+    await reloadedMeeting.save();
+    return mailCampaign;
   }
 
   async sendMailCampaign(id) {
-    try {
-      return ajax({
-        method: 'POST',
-        url: `/newsletter/sendMailCampaign/${id}`,
-      });
-    } catch (error) {
-      console.warn('An exception ocurred: ', error);
+    const endpoint = `/newsletter/send-mail-campaign`;
+    const body = {
+      data: {
+        id: id,
+      },
+    };
+    const response = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/vnd.api+json',
+      },
+      body: JSON.stringify(body),
+    });
+    if (!response.ok) {
       this.toaster.error(
         this.intl.t('error-send-newsletter'),
         this.intl.t('warning-title')
       );
-      return null;
+      throw new Error('An exception ocurred: ' + response.error);
```

Comment:
Dit zit toch niet helemaal goed.
Ik heb lokaal een mail campagne gemaakt die niet verstuurd zou mogen worden.
(1 kort bestek van een nota zonder een thema geselecteerd)
De campaign aanmaken zal lukken, maar het versturen zal resulteren in een error van mailchimp:
"Error: Your Campaign is not ready to send." in de newsletter-service.
Echter krijgt de frontend wel een 200 binnen
![image](https://user-images.githubusercontent.com/22245223/146954850-71a58c49-3176-4683-98ef-b8b2737e6899.png)
Maar de response is wel degelijk een error.
![image](https://user-images.githubusercontent.com/22245223/146954962-2d748615-a8de-4b8a-812a-1b46b0c736aa.png)

De frontend denkt dus onterecht dat de campagne is verstuurd, en we setten 'isSent' waardoor opnieuw versturen niet mogelijk is.

Reply:
Wat er lijkt te gebeuren is dat de fetch call al resolved is voor de response er is.
Want je krijgt tussentijds een andere response binnen (status ok, code 200, type 'basic') als je logt vlak na de fetch call.

Reply:
rand info, de reden dat 1 nota zonder thema's faalt is omdat er dan 0 ontvangers zijn en mailchimp is zo slim om dan het verzenden van die nieuwsbrief tegen te houden